packages/external-db-firestore/lib/firestore_utils.js

```diff
@@ -1,28 +1,30 @@
-// const { escapeId } = require('mysql')
-// const { InvalidQuery } = require('velo-external-db-commons').errors
-
-// const recordSetToObj = (rows) => rows.map(row => row.toJSON())
-//
-// const patchFieldName = (f) => {
-//     if (f.startsWith('_')) {
-//         return `x${f}`
-//     }
-//     return f
-// }
-//
-// const unpatchFieldName = (f) => {
-//     if (f.startsWith('x_')) {
-//         return f.slice(1)
-//     }
-//     return f
-// }
-//
-// const testLiteral = s => /^[a-zA-Z_0-9_]+$/.test(s)
-// const validateLiteral = l => {
-//     if (!testLiteral(l)) {
-//         throw new InvalidQuery(`Invalid literal`)
-//     }
-//     return `@${l}`
-// }
-
-module.exports = { /*recordSetToObj, escapeId, patchFieldName, unpatchFieldName, testLiteral, validateLiteral*/ }
+const { Timestamp } = require('@google-cloud/firestore')
+
+
+const fixDates = (value) => {
+    if (value instanceof Timestamp) {
+        return value.toDate()
+    }
+    return value
+}
+
+const deleteQueryBatch = async (db, query, resolve) => {
```

Comment:
this code cannot be here, it should be on the main impl firestore_data_provider.js